titan-compiler/parser.lua

```diff
@@ -45,6 +41,22 @@ function defs.boolopt(x)
     return x ~= ''
 end
 
+function defs.nil_exp(--[[ s ]])
+    -- We can't call Exp_Nil directly in the parser because we
+    -- need to drop the string capture that comes by default.
+    return ast.Exp_Nil()
+end
+
+function defs.number_exp(n)
+    if math.type(n) == 'integer' then
+        return ast.Exp_Integer(n)
+    elseif math.type(n) == 'float' then
+        return ast.Exp_Float(n)
+    else
+        error("impossible")
+    end
+end
+
```

Comment:
I still believe that we might be able to do this down to the lexical level and remove the dependency of `math.type`. Please, check out this [commit](https://github.com/titan-lang/titan-compiler/commit/efb5934d9489878159176249a1719d4734ae6df6) I made in the old repository.

Reply:
Doesn't separating things in the lexer make it harder to flag stuff like `4..` and `1337require` as syntax errors (as opposed to a sequence of two tokens)?